aws-modules/aws-s3/src/main/java/com/baeldung/s3/S3BucketOperationService.java

```diff
@@ -0,0 +1,60 @@
+package com.baeldung.s3;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.Bucket;
+import software.amazon.awssdk.services.s3.model.ListBucketsResponse;
+import software.amazon.awssdk.services.s3.model.NoSuchBucketException;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+class S3BucketOperationService {
+
+    private final S3Client s3Client;
+
+    public S3BucketOperationService(S3Client s3Client) {
+        this.s3Client = s3Client;
+    }
+
+    public boolean bucketExists(String bucketName) {
+        try {
+            s3Client.headBucket(request -> request.bucket(bucketName));
+            return true;
+        } catch (NoSuchBucketException exception) {
```

Comment:
Yes, ideally it would've been better if there was an API for this.
Calling the HEAD bucket operation is the accepted work around currently. Spring Cloud AWS uses it as well for their objectExists() implementation.